packages/key-management/src/util/ownSignatureKeyPaths.ts

```diff
@@ -59,6 +59,32 @@ const getStakingKeyPaths = (
   return paths;
 };
 
+/**
+ * Predicate that evaluates whether the given input set contains inputs that do not belong to the wallet.
+ *
+ * @param inputs The set of inputs to inspect.
+ * @param knownAddresses The wallet known address.
+ * @param inputResolver the input resolver.
+ */
+export const hasForeignInputs = async (
+  inputs: Cardano.TxIn[],
+  knownAddresses: GroupedAddress[],
+  inputResolver: Cardano.util.InputResolver
```

Comment:
Let's be explicit here (and in `ownSignatureKeyPaths` signature), because we'll be adding an input resolver that uses cardano-services provider. In the context of determining key paths, we're only interested in our own inputs, which can be resolved using data that the wallet already has (not making any extra requests).
```suggestion
  ownInputResolver: Cardano.util.InputResolver
```